test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -144,6 +144,57 @@ void performFailoverForTestsInvolvingTxns() throws ExecutionException {
                 testState.getEventReadCount(), testState.getEventWrittenCount());
     }
 
+    void performFailoverForTestsInvolvingKVS() throws ExecutionException {
+        long kvpEntryCount = getListKvp();
+        log.info("Current insert KVP count {}", kvpEntryCount);
+        log.info("ScaleUp one Segmentstore and Controller");
+
+        // Case-64 Scale up SegmentStore one instances from 3 to 4
+        Futures.getAndHandleExceptions(segmentStoreInstance.scaleService(4), ExecutionException::new);
+        // Case-60 Scale up controller one instances from 2 to 3
+        Futures.getAndHandleExceptions(controllerInstance.scaleService(3), ExecutionException::new);
+        log.info("Successfully scale up Controller 2 -> 3 and Segmentstore 3 -> 4");
+        log.info("Sleeping for {} ", WAIT_AFTER_FAILOVER_MILLIS);
+        Exceptions.handleInterrupted(() -> Thread.sleep(WAIT_AFTER_FAILOVER_MILLIS));
+        assertTrue("Current KVP entry count is not greater than previousKVPCount", kvpEntryCount < getListKvp());
+        kvpEntryCount = getListKvp();
+        log.info("Insert KVP count: {} without any failover after sleep before scaling", kvpEntryCount);
+        log.info("ScaleUp more than one Segmentstore and Controller");
+
+        // Case-65 Scale up SegmentStore to 2 instances from 4 to 6
+        Futures.getAndHandleExceptions(segmentStoreInstance.scaleService(6), ExecutionException::new);
```

Comment:
6 instances seems too much, as we may not have sufficient cluster resources to support all of them. Why don't you follow the same structure of scale up and down as the other failover tests?

Reply:
Resolved, scale up max to 3 instance